radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/PeerEvent.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * (C) Copyright 2021 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.network.p2p;
+
+import com.radixdlt.network.p2p.transport.PeerChannel;
+
+import java.util.Objects;
+
+public interface PeerEvent {
```

Comment:
Some javadoc would be helpful, what's the purpose of this marker interface? Why not just treat each PeerEvent as a separate class without a shared marker interface?

Reply:
just so we can group related events into a single event dispatcher/processor